### ISSUE TITLE
Support selecting audio-output device

### ISSUE THREAD

**Issue by SimonSimCity** [bcc-code/bmm-web#137]
We have not yet spend any time to support selecting an audio-output device. Here's some feedback I got from @Ghostbird (Linux):

> Furthermore, now that I've got it as a more integrated app, I'd like UI to switch which audio sink is used to play the music. You can get a list of those with the call: `await navigator.mediaDevices.enumerateDevices().then(devices => devices.filter((device) => device.kind === 'audiooutput'))`

> [audio-sinks.txt](https://github.com/bcc-code/bmm-web/files/11633313/audio-sinks.txt) On my system that outputs this, but note that I've got a complex audio set-up compared to the average listener. Most people will have a list that's less than half as long.
> I've checked and verified that the `deviceId` property in that list is pretty stable. I haven't seen it change, even when disconnecting and reconnecting completely virtual audio devices that are loaded by JACK modules from a script. So you could store the chosen `deviceId` as local state to restore it on next app startup. You can `await media.setSinkId(deviceId);` to select a sink, where `media` is the `HTMLMediaElement` of the player.

**Comment by kkuepper** [comment]
I don't see how this is relevant on Windows or Mac. That's handled by the operating system.

**Comment by Ghostbird** [comment]
I don't agree that it's handled by the Operating System. I think my set-up is more flexible than the average user's set-up, and yet I can't properly set it on operating system. The underlying reason is that from my OS' sound management perspective, all electron and PWA apps are instances of Chromium. You can select individual audio devices for them on OS level, but those aren't properly stored and reapplied on restart. Microsoft Teams PWA has internal selection of which audio output you want to use. I assume that's because this also applies on Windows, otherwise Microsoft wouldn't have added that.

However, this is a really _low priority_ thing.

**Comment by kkuepper** [comment]
You're comparing a messenger/video-conferencing app to a music player. Since Spotify doesn't offer audio device selection it's not really needed for a music player. But obviously that's a completely different picture for a messaging app.

Also, since this is a community project, if you want a very niche feature you can just build it. As long as it doesn't over-complicate things for other users, it will be added.
